### PR TITLE
Add buf.registry.legacy.federation.v1beta1 package

### PR DESCRIPTION
**Pull request by bufdev**
This adds the `buf.registry.legacy.federation.v1beta1` package.

We need to support legacy federation for a few of our early customers, and this is the easiest way for us to do so. It effectively means supporting federation on two endpoints: `GetGraph` and `Upload`. The edits to these endpoints and their associated types will mean an API that is sufficiently different and unclean that we instead are splitting these out into their own package (and associated endpoints).

### CODE REVIEW

**Review by saquibmian — COMMENTED** [review]
Don't you also need a `GraphService` in the legacy package?

File: buf/registry/legacy/federation/v1beta1/upload_service.proto [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    // be set, and setting it is an error.
+    string commit_id = 2 [(buf.validate.field).string.uuid = true];
+    // The registry hostname of the dep.
+    string registry = 3 [
+      (buf.validate.field).required = true,
+      (buf.validate.field).string.hostname = true
+    ];
+  }
+  // Content to upload for a given reference.
+  message Content {
```
[the remark this answers is not in the document]
Comment (bufdev) [reply buf/registry/legacy/federation/v1beta1/upload_service.proto:61]: You can't upload content for other registries correct, only depend on them.

File: buf/registry/module/v1beta1/graph.proto [hunk 2]
```diff
@@ -26,18 +26,21 @@ option go_package = "buf.build/gen/go/bufbuild/registry/protocolbuffers/go/buf/r
...
       (buf.validate.field).required = true,
       (buf.validate.field).string.uuid = true
     ];
+  }
+
+  // An edge in the dependency graph.
+  message Edge {
+    // The Node of the start of the edge.
+    Node from_node = 1 [(buf.validate.field).required = true];
     // The Commit of the end of the edge.
```
[the remark this answers is not in the document]
Comment (bufdev) [reply buf/registry/module/v1beta1/graph.proto]: done

File: buf/registry/legacy/federation/v1beta1/graph.proto [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      (buf.validate.field).required = true,
+      (buf.validate.field).string.hostname = true
+    ];
+  }
+
+  // An edge in the dependency graph.
+  message Edge {
+    // The Node of the start of the edge.
+    Node from_node = 1 [(buf.validate.field).required = true];
+    // The Commit of the end of the edge.
```
[the remark this answers is not in the document]
Comment (bufdev) [reply buf/registry/legacy/federation/v1beta1/graph.proto]: done

File: buf/registry/legacy/federation/v1beta1/graph.proto [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  message Node {
+    // The commit of the node.
+    string commit_id = 1 [(buf.validate.field).required = true];
+    // The registry hostname of the Node.
+    string registry = 2 [
+      (buf.validate.field).required = true,
+      (buf.validate.field).string.hostname = true
+    ];
+  }
+
```
[the remark this answers is not in the document]
Comment (bufdev) [reply buf/registry/legacy/federation/v1beta1/graph.proto]: done

**Review by saquibmian — APPROVED** [review]

**Pull request merged**